tests/options/tcoalesce.nim

```diff
@@ -0,0 +1,154 @@
+import options, unittest
+
+suite "coalesce":
```

Comment:
The tests should be in a single ``toptions`` module inside ``tests/stdlib`` (or at the bottom of the ``options`` module).